sessions/tracker.go

```diff
@@ -0,0 +1,101 @@
+package sessions
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+const (
+	//contextSessionKey is a unique key for accessing and setting Bugsnag
+	//session data on a context.Context object
+	contextSessionKey ctxKey = 1
+)
+
+// ctxKey is a type alias that ensures uniqueness as a context.Context key
+type ctxKey int
+
+// SessionTracker exposes a method for starting sessions that are used for
+// gauging your application's health
+type SessionTracker interface {
+	StartSession(context.Context) context.Context
+	GetSession(context.Context) *Session
+}
+
+type sessionTracker struct {
+	sessionChannel chan *Session
+	sessions       []*Session
+	config         *SessionTrackingConfiguration
+	publisher      sessionPublisher
+}
+
+// NewSessionTracker creates a new SessionTracker based on the provided config,
+func NewSessionTracker(config *SessionTrackingConfiguration) SessionTracker {
+	publisher := publisher{
+		config: config,
+		client: &http.Client{Transport: config.Transport},
+	}
+	st := sessionTracker{
+		sessionChannel: make(chan *Session, 1),
+		sessions:       []*Session{},
+		config:         config,
+		publisher:      &publisher,
+	}
+	go st.processSessions()
+	return &st
+}
+
+func (s *sessionTracker) GetSession(ctx context.Context) *Session {
+	return ctx.Value(contextSessionKey).(*Session)
```

Comment:
I think this will crash if someone forgets to add the session to the context thats passed into the autonotify